packages/react/src/executors/module-federation-dev-server/module-federation-dev-server.impl.ts

```diff
@@ -147,52 +133,139 @@ export default async function* moduleFederationDevServer(
       )
     );
   }
+  return devRemoteIters;
+}
+
+async function buildStaticRemotes(
+  remotes: {
+    remotePorts: any[];
+    staticRemotes: string[];
+    devRemotes: string[];
+  },
+  nxBin,
+  context: ExecutorContext,
+  options: ModuleFederationDevServerOptions
+) {
+  logger.info(`NX Building ${remotes.staticRemotes.length} static remotes...`);
+  const mappedLocationOfRemotes: Record<string, string> = {};
+
   for (const app of remotes.staticRemotes) {
-    const remoteProjectServeTarget =
-      context.projectGraph.nodes[app].data.targets['serve-static'];
-    const isUsingModuleFederationDevServerExecutor =
-      remoteProjectServeTarget.executor.includes(
-        'module-federation-dev-server'
-      );
-    let outWithErr: null | string[] = [];
+    mappedLocationOfRemotes[app] = `http${options.ssl ? 's' : ''}://${
+      options.host
+    }:${options.staticRemotesPort}/${app}`;
+  }
+
+  process.env.NX_MF_DEV_SERVER_STATIC_REMOTES = JSON.stringify(
+    mappedLocationOfRemotes
+  );
+
+  await new Promise<void>((res) => {
     const staticProcess = fork(
       nxBin,
       [
-        'run',
-        `${app}:serve-static${
-          context.configurationName ? `:${context.configurationName}` : ''
-        }`,
-        ...(isUsingModuleFederationDevServerExecutor
-          ? [`--isInitialHost=false`]
+        'run-many',
+        `--target=build`,
+        `--projects=${remotes.staticRemotes.join(',')}`,
+        ...(context.configurationName
+          ? [`--configuration=${context.configurationName}`]
           : []),
+        ...(options.parallel ? [`--parallel=${options.parallel}`] : []),
       ],
       {
         cwd: context.root,
         stdio: ['ignore', 'pipe', 'pipe', 'ipc'],
       }
     );
     staticProcess.stdout.on('data', (data) => {
-      if (isCollectingStaticRemoteOutput) {
-        outWithErr.push(data.toString());
-      } else {
-        outWithErr = null;
+      const ANSII_CODE_REGEX =
+        /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g;
+      const stdoutString = data.toString().replace(ANSII_CODE_REGEX, '');
+      if (stdoutString.includes('Successfully ran target build')) {
```

Comment:
Maybe something for later since this logic of checking `stdout` is in existing code. Maybe Nx should be sending a message when parent spawns it with IPC. Like `process.send && process.send('nx.run.success')` or something.

The text may change, but as long as we have good e2e coverage it's okay for now.